### PR TITLE
validate-modules: Documentation bool

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This check allows to catch cases where type of argument is different than documentation does.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
docs/docsite/rst/dev_guide/testing_validate-modules.rst
test/sanity/validate-modules/main.py

### CODE REVIEW

File: test/sanity/validate-modules/main.py [hunk 1]
```diff
@@ -1229,11 +1229,26 @@ def _validate_argument_spec(self, docs, spec, kwargs):
...
+                    self.reporter.error(
+                        path=self.object_path,
+                        code=325,
+                        msg='argument_spec for "%s" defines type="bool" but documentation does not' % (arg,)
+                    )
+
+                if data['type'] != doc_type:
+                    self.reporter.error(
+                        path=self.object_path,
+                        code=336,
```
Comment (dagwieers) [inline test/sanity/validate-modules/main.py]: IMO the previous block is not needed, and this error should become error 325.

**Review by dagwieers — CHANGES_REQUESTED** [review]

File: test/sanity/validate-modules/main.py [hunk 2]
```diff
@@ -1229,11 +1229,26 @@ def _validate_argument_spec(self, docs, spec, kwargs):
...
+                        msg='argument_spec for "%s" defines type as %s but documentation defines type as %s' % (arg, data['type'], doc_type)
+                    )
+
+            if 'type' not in data and doc_type in frozenset(('list', 'dict', 'bool', 'int', 'float')):
```
Comment (felixfontein) [inline test/sanity/validate-modules/main.py]: Shouldn't there also be `path` and some more in this list?

Comment (dagwieers) [reply test/sanity/validate-modules/main.py]: That list should be shared with the Schema tests :)

Comment (dagwieers) [reply test/sanity/validate-modules/main.py]: But to be honest, the types should be identical, it shouldn't matter what it is.

**Comment by dagwieers** [comment]
skip_bot

**Review by dagwieers — APPROVED** [review]

**Comment by dagwieers** [comment]
I moved the module fixes to https://github.com/ansible/ansible/pull/52182

**Review by mattclay — APPROVED** [review]

**Pull request merged**